### PR TITLE
update tox.ini

### LINKED ISSUE #1239: Clean up tox testenvs
**Is your feature request related to a problem? Please describe.**
Some test environments might be out-to-date:
- Currently, we treat Python 3.8 builds in a different way than the previous Python versions. That was due because we introduced Python 3.8 as soon as it has been released, and some dependencies were not up-to-date.
- many libraries for linting might be updated to the latest version.
- for some test environments (especially linting), we don't need to install the package. Use `skipdist=True` where possible.
- in `workflow.yml` we still install `pipenv`, although it's not used anymore in `tox`.

**Describe the solution you'd like**
Address the above.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

- Skip the distribution for some testenvs when possible to save time (e.g. 'flake8', 'mypy' etc.)
- simplify test envs for main Python tests.

## Fixes

Fixes #1239 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
n/a

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -33,73 +30,24 @@ deps =
     pytest-rerunfailures==9.0
 
 commands =
-    pip install git+https://github.com/pytoolz/cytoolz.git#egg=cytoolz==0.10.1.dev0
+    ; ignore exit code of tensorflow because it does not work on Python 3.8
```
Comment (DavidMinarsch) [inline tox.ini:33]: It might be worth checking if we can upgrade the ml example to 2.+ of tensorflow without any work

File: tox.ini [hunk 2]
```diff
@@ -1,46 +1,14 @@
...
-envlist = bandit-main, bandit-tests, black, black-check, copyright_check, docs, flake8, liccheck, mypy, py3.8, py3.7, py3.6
-skipsdist = False
-ignore_basepython_conflict = True
+envlist = bandit, black, black-check, copyright_check, docs, flake8, liccheck, mypy, py{3.6,3.7,3.8}
 
 [testenv]
+basepython = python3
+skipsdist = False
+usedevelop = False
+skip_install = False
```
Comment (marcofavorito) [inline tox.ini:8]: by default, we:
- don't skip the creation of the source distribution (the output of `python setup.py sdist`)
- don't use `develop` installation
- don't skip install.

They were the default settings; but migth be good to state them explicitly for self-documentation. Thoughts? O/w if we think it's too verbose we can get rid of them.

File: tox.ini [hunk 3]
```diff
@@ -1,46 +1,14 @@
...
 whitelist_externals = /bin/sh
-
-[testenv:py3.8]
-basepython = python3.8
-passenv = *
-deps =
-    Cython
```
Comment (marcofavorito) [inline tox.ini:13]: this has been excluded because not needed anymore (it was needed for `cytoolz`)

File: tox.ini [hunk 4]
```diff
@@ -1,46 +1,14 @@
...
-    black==19.10b0
-    mistune==2.0.0a4
-    aiohttp==3.6.2
-    SQLAlchemy==1.3.16
-    pynacl==1.3.0
-    pexpect==4.8.0
-    pytest-rerunfailures==9.0
-
-commands =
-    pip install git+https://github.com/pytoolz/cytoolz.git#egg=cytoolz==0.10.1.dev0
```
Comment (marcofavorito) [inline tox.ini:36]: We are at `0.10.1` now: https://pypi.org/project/cytoolz/

File: tox.ini (hunk 2 above)
Comment (marcofavorito) [reply tox.ini:8]: We could use `develop` in the main testenv to save some time, but better to test also the packaging.

File: tox.ini [hunk 5]
```diff
@@ -64,72 +32,65 @@ deps =
...
-commands =
-    pip install .[all]
-    pip install -i https://test.pypi.org/simple/ fetch-p2p-api==0.0.2
-    pytest -rfE --doctest-modules aea packages/fetchai/protocols packages/fetchai/connections tests/ --cov-report=html --cov-report=xml --cov-report=term --cov=aea --cov=packages/fetchai/protocols --cov=packages/fetchai/connections {posargs}
+[testenv:py3.8]
+basepython = python3.8
 
 [testenv:bandit]
+skipsdist = True
+skip_install = True
```
Comment (marcofavorito) [inline tox.ini:49]: here we skip both the source distribution and the installation.

File: tox.ini [hunk 6]
```diff
@@ -64,72 +32,65 @@ deps =
...
 deps = bandit==1.6.2
 commands = bandit -s B101 -r aea benchmark examples packages scripts tests
 
 [testenv:black]
+skipsdist = True
+skip_install = True
```
Comment (marcofavorito) [inline tox.ini:55]: see above re. `bandit`

File: tox.ini [hunk 7]
```diff
@@ -64,72 +32,65 @@ deps =
...
 deps = black==19.10b0
 commands = black aea benchmark examples packages scripts tests
 
 [testenv:black-check]
+skipsdist = True
+skip_install = True
```
Comment (marcofavorito) [inline tox.ini:61]: see above

File: tox.ini [hunk 8]
```diff
@@ -64,72 +32,65 @@ deps =
...
+skipsdist = True
+usedevelop = True
+deps = ipfshttpclient
 commands = {toxinidir}/scripts/generate_ipfs_hashes.py --check {posargs}
 
 [testenv:package_version_checks]
-deps = python-dotenv
+skipsdist = True
+usedevelop = True
+deps =
```
Comment (marcofavorito) [inline tox.ini:81]: the only dependency is the `aea` package, installed with source installation via `usedevelop = True` and `skip_install = False` (by default)

**Comment by marcofavorito** [comment]
notably, after the latest changes, we got rid of `pip install .[all]` in the `commands:` list of the test envs.

File: tox.ini (hunk 2 above)
Comment (DavidMinarsch) [reply tox.ini:8]: I think we can document in the tox.ini as a comment. We should not use develop. I prefer to catch packaging issues early.

**Pull request merged**